src/modules/sensors/sensors.cpp

```diff
@@ -613,6 +643,15 @@ void Sensors::Run()
 		}
 	}
 
+	if (_hil_enabled) {
```

Comment:
Is some "failure detector" actually supposed to run also in  flying drone in  the end? Or is this framework only for simulation.

That is, should you actually initialize _failureDetector to a "real one" if not in hitl and to "failure_detector_HITL" when in hitl mode, and just call the update() always? If the intention is to actually inject a failure detector to the flying drone as well, that is.

Maybe make a base class for FailureDetector, inherit the failure_detector_hitlf from that, and use a pointer to "FailureDetector" base to call the initialized failure detector's update? To have a framework of adding a failure detector for the sensors.

Just ideas; the point here is, that if the failure detector is never supposed to run in a flying drone, then you probably shouldn't have the isBaroOk, isMagOk, isGpsOK calls when not in HITL in the sensors module?

Reply:
Thanks a lot for your advice with base class.  Actually we planed use it only in hitl mode for the reason of safety fly, however  it may be required in the future like for 'drone test stand'. So I suppose would be better move step by step and for being just introduce failure injection for hitl testing and then if it  is required, redesign it to be more elegant